scripts/hal_alignment/test_hal_gab_access.pl

```diff
@@ -0,0 +1,268 @@
+
+# Get MLSS obejct by ID:
+our $cactus_mlss = $mlss_adaptor->fetch_by_dbID($mlss_id);
+
+# Load test regions:
+my $test_regions =  load_regions($regions_tsv);
+
+my $res = {};
+
+# Test access via a slice:
+sub test_slice_access {
+    my $reg = shift;
+    my $res = shift;
+
+    my $sliceAdaptor = $registry->get_adaptor($reg->{Species}, 'Core', 'Slice');
+    my $slice = $sliceAdaptor->fetch_by_region($reg->{CoordSys}, $reg->{DnaFrag}, $reg->{Start}, $reg->{End});
+
+    my @slice_gabs = @{$gab_adaptor->fetch_all_by_MethodLinkSpeciesSet_Slice($cactus_mlss, $slice)};
+
+    my $slice_gab_count = scalar(@slice_gabs);
+    print "In region $reg->{Region} found $slice_gab_count GABs\n";
+    $res->{SliceSummary}->{GabCount}->{$slice_gab_count}++;
+
+    foreach my $i (0 .. $#slice_gabs) {
+        my $gab = $slice_gabs[$i];
+        $res->{SliceSummary}->{GabLength}->{$gab->length}++;
+    }
+}
+
+# Test pairwise access via dnafrags:
+sub test_dnafrag_access {
+    my $reg = shift;
+    my $res = shift;
+
+    my $min_gab_len_cutoff = int(abs($reg->{End} - $reg->{Start}) / 1000);
+    my $min_ga_len_cutoff  = $min_gab_len_cutoff / 4;
+
+	print "Testing pairwise access on region $reg->{Region}\n";
+
+    my $genome_db1 = $gdb_adaptor->fetch_by_name_assembly($reg->{Species});
+    my $dnafrag1 = $dnafrag_adaptor->fetch_by_GenomeDB_and_name($genome_db1, $reg->{DnaFrag});
+    my @dnafrag_gabs = @{$gab_adaptor->fetch_all_by_MethodLinkSpeciesSet_DnaFrag($cactus_mlss, $dnafrag1, $reg->{Start}, $reg->{End})};
+    
+    my $dnafrag2;
+    my $greatest_ga_dnafrag_length = 0;
+    foreach my $i (0 .. $#dnafrag_gabs) {
+        my $gab = $dnafrag_gabs[$i];
+        if ($gab->length >= $min_gab_len_cutoff) {
+            foreach my $ga (@{$gab->get_all_non_reference_genomic_aligns}) {
+                my $ga_dnafrag_length = $ga->dnafrag_end - $ga->dnafrag_start + 1;
+                if ($ga_dnafrag_length >= $min_ga_len_cutoff && $ga_dnafrag_length > $greatest_ga_dnafrag_length) {
+                    $greatest_ga_dnafrag_length = $ga_dnafrag_length;
+                    $dnafrag2 = $ga->dnafrag;
+                }
+            }
+        }
+    }
+
+	if (defined $dnafrag2) {
+
+        my $genome_db2 = $gdb_adaptor->fetch_by_name_assembly($dnafrag2->genome_db->name);
+        $dnafrag2 = $dnafrag_adaptor->fetch_by_GenomeDB_and_name($genome_db2, $dnafrag2->name);
+
+        my @dnafrag2_gabs = @{$gab_adaptor->fetch_all_by_MethodLinkSpeciesSet_DnaFrag_DnaFrag($cactus_mlss, $dnafrag1, $reg->{Start}, $reg->{End}, $dnafrag2)};
+
+        my $dnafrag2_gab_count = scalar(@dnafrag2_gabs);
+    	$res->{PairwiseSummary}->{GabCount}->{$dnafrag2_gab_count}++;
+
+        foreach my $i (0 .. $#dnafrag2_gabs) {
+            my $gab = $dnafrag2_gabs[$i];
+			$res->{PairwiseSummary}->{GabLength}->{$gab->length}++;
+        }
+	}
+ 
+
+}
+
+# Run tests on all regions:
+for my $reg (@$test_regions) {
+   test_slice_access($reg, $res); 
+   test_dnafrag_access($reg, $res);
```

Comment:
Leading and trailing whitespace:
```suggestion
    test_slice_access($reg, $res);
    test_dnafrag_access($reg, $res);
```